src/components/scenes/Staking/EarnScene.tsx

```diff
@@ -0,0 +1,168 @@
+                const { staked, earned, unstaked } = allocations
+                if ([...staked, ...earned, ...unstaked].some(positionAllocation => !zeroString(positionAllocation.nativeAmount))) {
+                  positionWallets.push(wallet)
+                }
+              }
+            }
+
+            policyMap[wallet.id] = { stakePolicyPositions, stakePlugins }
+            setStakePolicyMap({ ...policyMap })
+            setPositionWallets(positionWallets)
+          } catch (e) {
+            console.error(String(e))
+          }
+        }
+      }
+      setIsLoading(false)
+    },
+    [],
+    'EarnScene'
+  )
+
+  // Filter wallets based on isPortfolioSelected
+  React.useEffect(() => {
+    if (stakePolicyMap == null) return
+
+    if (isPortfolioSelected) {
+      setDisplayWallets(positionWallets)
+    } else {
+      setDisplayWallets(wallets)
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [isPortfolioSelected, stakePolicyMap])
+
+  const renderStakeItems = React.useCallback(
+    (item: FlatListItem<EdgeCurrencyWallet>) => {
+      const { item: wallet } = item
+      if (stakePolicyMap == null) return <></>
+
+      const { stakePolicyPositions, stakePlugins } = stakePolicyMap[wallet.id] ?? { stakePolicyPositions: [], stakePlugins: [] }
+
+      return (
+        <>
+          {stakePolicyPositions.map((stakePolicyPosition: { stakePolicy: StakePolicy; stakePosition: StakePosition }, index: number) => {
+            const { stakePolicy, stakePosition } = stakePolicyPosition
+
+            if (stakePolicy == null) return <></>
+            const stakePlugin = getPluginFromPolicy(stakePlugins, stakePolicy)
+
+            const handlePress =
+              stakePlugin == null ? undefined : () => navigation.push('stakeOverview', { stakePlugin, walletId: wallet.id, stakePolicy, stakePosition })
+
+            return (
+              <EdgeAnim key={`${wallet.id}-${index}`} enter={fadeInUp20}>
+                <StakingReturnsCard wallet={wallet} stakePolicy={stakePolicy} isOpenPosition={isPortfolioSelected} onPress={handlePress} />
+              </EdgeAnim>
+            )
+          })}
+        </>
+      )
+    },
+    [navigation, stakePolicyMap, isPortfolioSelected]
+  )
+
+  return (
+    // TODO: Address "VirtualizedLists should never be nested inside plain
+    // ScrollViews with the same orientation because it can break windowing and
+    // other functionality - use another VirtualizedList-backed container
+    // instead." somehow, while retaining the bottom loader positioning...
+    <SceneWrapper scroll padding={theme.rem(0.5)}>
+      <EdgeSwitch labelA={lstrings.staking_discover} labelB={lstrings.staking_portfolio} onSelectA={handleSelectEarn} onSelectB={handleSelectPortfolio} />
+      <SectionHeader leftTitle={lstrings.staking_earning_pools} />
+      <FlatList data={displayWallets} renderItem={renderStakeItems} keyExtractor={wallet => wallet.id} style={styles.list} />
```

Comment:
Instead of using a `FlatList`, why not render the items directly? Since you've placed the list inside a scroll view already, the `FlatList` won't be able to optimize rendering in any case.

We shouldn't have 100's of options, so it shouldn't be a long list.  It seems like a `{displayWallets.map(wallet => renderStakeItems(wallet))}` would basaically do the same job.